hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/meta/CkpMetadata.java

```diff
@@ -97,8 +97,6 @@ public void close() {
   public void bootstrap(HoodieTableMetaClient metaClient) throws IOException {
     fs.delete(path, true);
     fs.mkdirs(path);
-    metaClient.getActiveTimeline().getCommitsTimeline().filterPendingExcludingCompaction()
-        .lastInstant().ifPresent(instant -> startInstant(instant.getTimestamp()));
```

Comment:
if we remove the logic of start last pending instant, how can we recommit the inflight commit? cc @wxplovecc @danny0405

Reply:
Would rollback it first before starting a new one.

Reply:
@danny0405 When flink checkpoint complete but hudi commit failed, we need recommit the commit, rollback the inflight commit will lose data, any new logic to guard this case?

Reply:
@leesf shall we consider this related PR？https://github.com/apache/hudi/pull/5913 
It will re-send pending instants to ckp_meta when bootstrap, and introduce a mechanism of CANCELLED state to eliminate rollbacked instants. 
Thus last successful instant will be kept when recovering for recommit